specification/connector_sync_job/_types/SyncJob.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Id } from '@_types/common'
+import { long } from '@_types/Numeric'
+import { Dictionary } from '@spec_utils/Dictionary'
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+import {
+  ConnectorConfiguration,
+  FilteringConfig,
+  IngestPipelineParams,
+  SyncStatus
+} from '../../connector/_types/Connector'
+
+interface SyncJobConnectorReference {
+  configuration: ConnectorConfiguration
+  filtering: FilteringConfig
+  id: Id
+  index_name: string
+  language?: string
+  pipeline: IngestPipelineParams
+  service_type: string
+}
+
+enum SyncJobType {
+  full,
+  incremental,
+  access_control
+}
+
+enum TriggerMethod {
+  on_demand,
+  scheduled
+}
+
+export interface ConnectorSyncJob {
+  cancelation_requested_at?: string
+  canceled_at?: string
+  completed_at?: string
+  connector: SyncJobConnectorReference
+  created_at: string
+  deleted_document_count: long
+  error?: string
+  id: Id
+  indexed_document_count: long
+  indexed_document_volume: long
+  job_type: SyncJobType
```

Comment:
can `job_type` be a SyncJobType or SyncJobType[] enum consistently?

Reply:
same for trigger_method

Reply:
In a `ConnectorSyncJob` representation they are not an array, there is singe value associated with `trigger_method` and `job_type`

Reply:
What I mean is, in this class `job_type` and `trigger_method` are enums, while in the classes SyncJobPostRequest.ts and SyncJobListRequest.ts they are `Name`/`Names`, which are just strings, can't they be the same enum everywhere?

Reply:
Ahh I see what you mean! For `job_type` this is passed as a comma-separated list in the URL param of `SyncJobListRequest`. I was not sure if defining this as `SyncJobType[]` would work with URL param list, and I saw `Names` in other examples of requests that pass comma separated lists in url params. 

If the usage of `SyncJobType[]` and `SyncJobTriggerMethod` is correct, I can adapt this in the PR.

Reply:
thanks :) I think it's more coherent this way from the clients perspective, to always provide enums where there's fixed values

Reply:
I addressed that feedback in the updated PR :)